stubs/qrcode/qrcode/image/pure.pyi

```diff
@@ -1,16 +1,22 @@
-from _typeshed import Incomplete
 from collections.abc import Generator
+from typing import IO, Any, Literal
+from typing_extensions import TypeAlias
 
-import qrcode.image.base
+from . import base
 
-class PyPNGImage(qrcode.image.base.BaseImage):
+# png.Writer; no types available
+_Writer: TypeAlias = Any
+
+class PyPNGImage(base.BaseImage):
     kind: str
-    allowed_kinds: Incomplete
-    needs_drawrect: bool
-    def new_image(self, **kwargs): ...
-    def drawrect(self, row, col) -> None: ...
-    def save(self, stream, kind: Incomplete | None = None) -> None: ...
-    def rows_iter(self) -> Generator[Incomplete, Incomplete, None]: ...
-    def border_rows_iter(self) -> Generator[Incomplete, None, None]: ...
+    allowed_kinds: tuple[Literal["PNG"]]
+    # the new_image and get_image methods accept arbitrary keyword arguments to
+    # accommodate subclasses with additional arguments.
+    def new_image(self, **kwargs: Any) -> _Writer: ...
+    def get_image(self, **kwargs: Any) -> _Writer: ...
+    def drawrect(self, row: int, col: int) -> None: ...
+    def save(self, stream: IO[bytes], kind: str | None = None) -> None: ...
```

Comment:
Looking at the source of the `png` module [here](https://gitlab.com/drj11/pypng/-/blob/main/code/png.py?ref_type=heads), it seems this could be `SupportsWrite` (from `_typeshed`):

```suggestion
    def save(self, stream: SupportsWrite[bytes], kind: str | None = None) -> None: ...
```